frontend/app/src/components/settings/accounting/rule/AccountingRuleSetting.vue

```diff
@@ -204,45 +155,120 @@ function showDeleteConfirmation(item: AccountingRuleEntry) {
   );
 }
 
-function getType(eventType: string, eventSubtype: string) {
-  return get(
-    getEventTypeData({
-      eventSubtype,
-      eventType,
-    }),
-  );
+async function handleRuleAction(action: 'add-general' | 'add-event-specific' | 'edit-general' | 'edit-event-specific') {
+  const props = get(actionDialogProps);
+  if (!props)
+    return;
+
+  const { eventId, eventSpecificRule, generalRule } = props;
+
+  switch (action) {
+    case 'add-general': {
+      const { query } = get(route);
+      const { counterparty, eventSubtype, eventType } = query;
+      add({
+        ...getPlaceholderRule(),
+        counterparty: counterparty?.toString() ?? null,
+        eventSubtype: eventSubtype?.toString() ?? '',
+        eventType: eventType?.toString() ?? '',
+      });
+      break;
+    }
+    case 'add-event-specific': {
+      const { query } = get(route);
+      const { counterparty, eventSubtype, eventType } = query;
+      add({
+        ...getPlaceholderRule(),
+        counterparty: counterparty?.toString() ?? null,
+        eventSubtype: eventSubtype?.toString() ?? '',
+        eventType: eventType?.toString() ?? '',
+      }, [eventId]);
+      break;
+    }
+    case 'edit-general':
+      if (generalRule)
+        edit(generalRule);
+      break;
+    case 'edit-event-specific':
+      if (eventSpecificRule)
+        edit(eventSpecificRule, eventSpecificRule.eventIds ?? undefined);
+      break;
+  }
+
+  set(actionDialog, false);
+  await router.replace({ query: {} });
 }
 
 onMounted(async () => {
   const { query } = get(route);
-  const { 'add-rule': addRule, counterparty, 'edit-rule': editRule, eventSubtype, eventType } = query;
+  const { 'add-rule': addRule, counterparty, 'edit-rule': editRule, eventId, eventSubtype, eventType } = query;
 
   const ruleData = {
     counterparty: counterparty?.toString() ?? null,
     eventSubtype: eventSubtype?.toString() ?? '',
     eventType: eventType?.toString() ?? '',
   };
 
-  async function openDialog(rule?: AccountingRuleEntry, forceAdd = false) {
-    if (rule) {
-      startPromise(nextTick(() => forceAdd ? add(rule) : edit(rule)));
-    }
-    await router.replace({ query: {} });
-  }
+  const eventIdNum = eventId ? Number(eventId) : undefined;
 
   if (addRule) {
-    await openDialog({
-      ...getPlaceholderRule(),
-      ...ruleData,
-    }, true);
+    if (eventIdNum) {
```

Comment:
would it be possible to avoid the nested if and have a function for addRule and a function for editRule that handle the details?